src/tigerbeetle/cli.zig

```diff
@@ -144,21 +145,16 @@ pub fn parse_args(allocator: std.mem.Allocator) !Command {
     var cache_grid: ?[]const u8 = null;
     var verbose: ?bool = null;
 
+    // This iterator owns the arguments and is passed to the caller as a part of `Command`.
     var args = try std.process.argsWithAllocator(allocator);
-    // NOTE: do not deinit() as we return the path and this could free/invalidate it.
-    // defer args.deinit();
-
-    // Keep track of the args from the ArgIterator above that were allocated
-    // then free them all at the end of the scope.
-    var args_allocated = std.ArrayList([:0]const u8).init(allocator);
+    errdefer args.deinit(allocator);
 
     // Skip argv[0] which is the name of this executable
     const did_skip = args.skip();
     assert(did_skip);
 
     const raw_command = args.next() orelse
         fatal("no command provided, expected 'start', 'format', or 'version'", .{});
-    defer allocator.free(raw_command);
```

Comment:
I think this slipped during upgrade, and its probably UB: the `raw_command` might be libc argv, which shouldn't be freed.